store/store_asserts.go

```diff
@@ -66,6 +76,7 @@ func (s *Store) Assertion(assertType *asserts.AssertionType, primaryKey []string
 		asrt, e = dec.Decode()
 		return e
 	}, func(svcErr *assertionSvcError) error {
+		// applicable to v1 errors only, no Status in v2 error response.
```

Comment:
v1 has "error_list" not "error-list" I think, so the code could be

```
(len(svcErr.ErrorList) > 0 && svcErr.ErrorList[0].Code == "not-found" /* v2 errors */ ) || svcErr.Status == 404 /* v1 */
```


no ?

Reply:
an option might also be to stick some of this logic to assertionSvcError as methods

Reply:
Ah yes, I missed "_" vs "-", thanks, that indeed simplifies the check.